### PR TITLE
LPD-27188 Invalidate WebDAV session when the user is deactivated

### PR DESCRIPTION
**Pull request by tomibiro**
[LPD-27188](https://liferay.atlassian.net/browse/LPD-27188)

### CODE REVIEW

**Comment by alvarosaugarlr** [comment]
ci:test:sf

**Comment by alvarosaugarlr** [comment]
ci:test:stable

**Comment by alvarosaugarlr** [comment]
ci:test:relevant

**Comment by alvarosaugarlr** [comment]
ci:test:security

**Review by alvarosaugarlr — COMMENTED** [review]
### Code

Validation: if the session user is deactivated, in such a case, the session is invalidated so that the contents cannot be accessed via WebDav. 

The validation was performed in
* digestAuth
* basicAuth
since these are the two authorizations accepted by the WebDav standard.

A small change in the variable names. 

### Test
Unit tests are made in which it is tested if the session is returned invalid or not (covers the change made). To cover all cases, where the session is not invalidated, two new cases have been added. In this way it is tested:
* digestAuth - with invalid session and with not invalid session.
* basicAuth - with session invalidated and with session not invalidated

**Comment by alvarosaugarlr** [comment]
run ci tests

**Comment by nikolettabuza** [comment]
ci:test:sf

**Comment by nikolettabuza** [comment]
ci:test:relevant

**Comment by stian-sigvartsen** [comment]
ci:test:relevant

**Comment by stian-sigvartsen** [comment]
ci:test:sf

**Comment by stian-sigvartsen** [comment]
ci:test:relevant

**Comment by stian-sigvartsen** [comment]
ci:test:relevant

**Comment by stian-sigvartsen** [comment]
I asked @nikolettabuza to push that commit (da62679) from another ticket's investigation so that we better comply with the Digest specification:
https://www.rfc-editor.org/rfc/rfc7235#section-1.1

>    Upon receipt of a request for a protected resource that omits
>    credentials, contains invalid credentials (e.g., a bad password) or
>    partial credentials (e.g., when the authentication scheme requires
>    more than one round trip), an origin server SHOULD send a 401
>    (Unauthorized) response that contains a WWW-Authenticate header field
>    with at least one (possibly new) challenge applicable to the
>    requested resource.

Without issuing a challenge here, the HTTP filter chain terminates and an arbitrary HTTP response/code is returned to the client.

Also with this change we are consistently responding with a challenge whenever we return null, which avoid user enumeration vulnerability.

**Comment by stian-sigvartsen** [comment]
ci:test:relevant

**Comment by stian-sigvartsen** [comment]
ci:abort

**Comment by stian-sigvartsen** [comment]
ci:test:sf

**Comment by stian-sigvartsen** [comment]
ci:test:relevant

**Comment by stian-sigvartsen** [comment]
The `BaseAuthFilterTest#testGivenDigestAuthenticationWhenUserIsDeactivatedThenHttpSessionInvalidated` failure is relevant

**Comment by stian-sigvartsen** [comment]
@evibudai will work on the test fix. Thanks Évi!

**Comment by stian-sigvartsen** [comment]
Just adding a note of an offline conversation. The new test is failing because of the `HttpAuthManagerUtil` invocations that were added in the recent addition da62679fcdca4c6a46ba7ad7095668fb9d53095e . It may need mocking. @tomibiro is exploring it.

cc @nikolettabuza

**Comment by tomibiro** [comment]
ci:test:relevant

**Comment by tomibiro** [comment]
ci:sf

**Comment by stian-sigvartsen** [comment]
ci:test:sf

**Comment by tomibiro** [comment]
I do not see relevant test failures

**Comment by manuelecastro** [comment]
ci:report:48055767

**Comment by stian-sigvartsen** [comment]
@tomibiro This is looking really good! 

One thing I found was that you had been careful to describe in the test method names which auth scheme the test is covering, yet you are only asserting that the session was invalidated. 
So you don't know which of the `BaseAuthFilter#basicAuth` / `BaseAuthFilter#digestAuth` methods were executed.

If you don't want to test which was executed then you only need 2/4 tests that you added. However since you've gone to great length to test both `BaseAuthFilter` methods, then I think you should also assert that the response challenge is the same scheme.

In summary you should follow through on one approach or another. I'll assign back to you to consider how to proceed.

**Review by stian-sigvartsen — CHANGES_REQUESTED** [review]
See my previous comment. Also I added some SF commits.

**Comment by stian-sigvartsen** [comment]
ci:test:sf

**Comment by stian-sigvartsen** [comment]
ci:test:relevant

**Comment by stian-sigvartsen** [comment]
@tomibiro please can you review the changes?

**Review by tomibiro — COMMENTED** [review]
nice solution :-)

**Comment by stian-sigvartsen** [comment]
ci:forward
